src/background.ts

```diff
@@ -299,6 +302,44 @@ ipcMainHandle("MAXIMIZE_WINDOW", () => {
   }
 });
 
+ipcMainHandle("RESTART_ENGINE", async () => {
+  /*
+    プロセスが生存している場合はexitCodeにnull、終了していればnumber型のexit codeが代入されています。
+    プロセスが既に落ちている場合にtreeKillを実行する意味がないのでこうしてあります。
+  */
+  if (engineProcess.exitCode !== null) {
+    runEngine();
+    return;
+  }
+
+  // エンジンエラー時のエラーウィンドウ抑制用。
+  willQuitEngine = true;
+
+  /*
+    「killに使用するコマンドが終了するタイミング」と「OSがプロセスをkillするタイミング」が違うので単純にtreeKillのコールバック関数でrunEngine()を実行すると失敗します。
+     closeイベントはexitイベントよりも後に発火します。
```

Comment:
スペース一個多いかもです（細かいですが気になってしまったので）
```suggestion
    closeイベントはexitイベントよりも後に発火します。
```

Reply:
修正しました．